### PR TITLE
Add backend calls for 'shutdown of services' step

### PR DESCRIPTION
**Pull request by jsuchome**
So, this is obviously wrong from UI perspective, handling of error from `dump_database` is not correct ... (  but maybe that could be solved in the final UI?)

At least the right calls to API are there..

### CODE REVIEW

File: crowbar_framework/app/controllers/setups_controller.rb [hunk 1]
```diff
@@ -22,7 +22,31 @@ def show
...
+      flash[:alert] = e.message
+      # FIXME: redirect to previous step?
+    end
+
+    # FIXME: now user needs to be told where is the database dump located in case
+    # he/she wants to manually fetch it
+    redirect_to setup_path
+  end
+
+  def finalize_stop
```
Comment (rsalevsky) [inline crowbar_framework/app/controllers/setups_controller.rb]: I would name it just `finalize` and the route in the routes.rb is missing. https://github.com/jsuchome/crowbar-core/blob/a444d3d6689496e188df103550f046b2b55e0a7b/crowbar_framework/config/routes.rb#L148

File: crowbar_framework/app/controllers/setups_controller.rb [hunk 2]
```diff
@@ -22,7 +22,31 @@ def show
...
+    # Stop services at the nodes
+    @service_object = CrowbarService.new logger
+
+    @service_object.shutdown_openstack_at_nodes
+
+    # Dump of the database could fail with the message that there's not enough space
+    # We need to show the error to user so he can manually prepare the space on node
+    # and than redo this step.
+    begin
+      @service_object.dump_database
```
Comment (rhafer) [inline crowbar_framework/app/controllers/setups_controller.rb]: Isn't the method called "dump_openstack_database" meanwhile?

Comment (jsuchome) [inline crowbar_framework/app/controllers/setups_controller.rb]: It is...

**Comment by rhafer** [comment]
> there's a new finalize_stop action which is not called from anywhere (I don't even know if it should be separate controller action)

ti should be called from button Step 6 I think.

**Comment by rhafer** [comment]
Looks good to me. CI is still failing because of the initial patch in not in D:C:6:S yet.

**Comment by rhafer** [comment]
+1

**Comment by nkrinner** [comment]
+1

**Pull request merged**